src/IAPWS_WaterProperties.f90

```diff
@@ -0,0 +1,4258 @@
+    d1 = d1v
+    IF (d2v <= DCrit) THEN
+      d2 = d2v
+    ELSE
+      d2 = DCrit
+    ENDIF
+  ELSE
+    d2 = d2v
+    IF (d1v >= DCrit) THEN
+      d1 = d1v
+    ELSE
+      d1 = DCrit
+    ENDIF
+  ENDIF
+
+  CALL WNPT3(d1, d2, p, t, eps, x)
+  DITER3 = x
+ENDFUNCTION diter3
+!
+!-------------------------------------------------------------------------------
+!TODO needs header
+FUNCTION NULLP3N(D, T, P)
+  REAL(SRK), INTENT(IN) :: D
+  REAL(SRK), INTENT(IN) :: T
+  REAL(SRK), INTENT(IN) :: P
+  REAL(SRK) :: NULLP3N
+
+  REAL(SRK) :: V
+
+  V = 1.0_SRK/D
+  NULLP3N = PVT3N(V, T) - P
+ENDFUNCTION nullp3n
+!
+!-------------------------------------------------------------------------------
+!TODO needs header
+SUBROUTINE WNPT3(XA, XB, P, T, EPS, X)
+  REAL(SRK), INTENT(IN) :: XA
+  REAL(SRK), INTENT(IN) :: XB
+  REAL(SRK), INTENT(IN) :: P
+  REAL(SRK), INTENT(IN) :: T
+  REAL(SRK), INTENT(IN) :: EPS
+  REAL(SRK), INTENT(OUT) :: X
+
+  REAL(SRK) :: X1, F1, F2, F3, X2, P1, P3, X3
+  INTEGER(SIK) :: irem, i
+
+  X1 = XA
+
+  F1 = nullp3n(X1, T, P)
+  X3 = XB
+  F3 = nullp3n(X3, T, P)
+
+  x = 0.0_SRK
+  DO I = 1, 40
+    IF (F1 /= F3) THEN
+      X = X1 + (X3 - X1)*F1/(F1 - F3)
+    ELSE
+      EXIT
+    ENDIF
+    IF (X < 0.0_SRK) X = (X1 + X3)/2.0_SRK
+    IF (ABS(X) < 1.E-8) THEN
+      IF (ABS(X - X1) < EPS) RETURN
+    ELSE
+      IF (ABS((X - X1)/X) < EPS) RETURN
+    ENDIF
+    F2 = nullp3n(X, T, P)
+    X2 = X1 - (X1 - X3)/2.0_SRK
+    P1 = F2*F1
+    P3 = F2*F3
+    IF ((P1 < 0.0_SRK) .AND. (P3 < 0.0_SRK)) THEN
+      P1 = ABS(P1)
+      P3 = ABS(P3)
+    ENDIF
+    IF (P1 <= P3) THEN
+      X3 = X1
+      F3 = F1
+    END IF
+    X1 = X
+    F1 = F2
+    IF ((X2 - X3)*(X2 - X1) >= 0.0_SRK) CYCLE
+    IF ((ABS(F1/F3)) <= (0.3_SRK)) THEN
+      X = X1 + ABS(F1/F3)*1.5_SRK*(X3 - X1)
+      IREM = 1
+    ELSE IF ((ABS(F1/F3)) >= (3.0_SRK)) THEN
+      IREM = 2
+    ELSE
+      X = (X1 + X3)/2.0_SRK
+      IREM = 0
+    ENDIF
+    F2 = nullp3n(X, T, P)
+    IF ((((F2*F1) >= 0.0_SRK) .AND. (IREM == 1)) .OR. (((F2*F3) >= 0.0_SRK) .AND. (IREM == 2))) THEN
+      X = (X1 + X3)/2.0_SRK
+      F2 = nullp3n(X, T, P)
+    ENDIF
+    IF (F2*F1 <= F2*F3) THEN
+      X3 = X1
+      F3 = F1
+    ENDIF
+    X1 = X
+    F1 = F2
+  ENDDO
+ENDSUBROUTINE WNPT3
+!
+!-------------------------------------------------------------------------------
+!TODO needs header
+FUNCTION tph3n(P, h)
+  REAL(SRK), INTENT(IN) :: P
+  REAL(SRK), INTENT(IN) :: h
+  REAL(SRK) :: tph3n
+
+  REAL(SRK) :: eps, TE
+  REAL(SRK) :: T1, T2, TG, X
+  INTEGER(SIK) :: IX, i2
+  REAL(SRK), PARAMETER :: &
+    hc = 2087.54684511650_SRK, &
+    a1 = 646.836937239618_SRK, &
+    a2 = 62.7217723183194_SRK, &
+    a3 = -18.0176207184787_SRK, &
+    a4 = 2.01708278241311_SRK, &
+    a5 = 247.531572315595_SRK, &
+    a6 = -41.1717709139789_SRK, &
+    a7 = -150.208861198343_SRK, &
+    a8 = 1431.77901876243_SRK, &
+    a9 = -862.734816079672_SRK, &
+    a0 = 49.9470109089612_SRK
```

Comment:
So continuation lines should be 4 spaces indented compared to the start of the line.  This occurs in quite a few places throughout the module.  If you can fix that without spending an inordinate amount of time, please do.

Reply:
I'm not sure I follow.  There are several places where they are trying to align digits, should all of those be 4 spaces and not aligh numbers?  I thought there was some lattitude to improve readability.

Reply:
Tried to align these things better.  DIdn't fully enforce 4 spaces, but when it made sense, I did, Also aligned decimal points for several large lists of parameters.

Reply:
Yeah, I'll take a look... to be more precise, continuation lines should be _at least_ 4 spaces.  I don't want to see less than that, but using more than that can certainly improve readability (sometimes I'll just add a couple spaces to the start of the line so that the continuation lines will align with it when using 4 spaces for them)

Reply:
I'm pretty good with how you did it.  The times you didn't do 4 spaces, it was always greater than 4 spaces and made sense to do something else.